Bard API Async Implementation for v0.1.16 compatible code

nishantchauhan949 opened the pull request:
Signed-off-by: Nishant [nishantchauhan949@hotmail.com](mailto:letsmailnishant@gmail.com)

THIS PROJECT IS IN MAINTENANCE MODE. We accept pull requests for Bug Fixes **ONLY**. NO NEW FEATURES ACCEPTED!

Bard API Async Implementation for v0.1.16 compatible code

## Description
Implemented a separate implementation of bardapi.core.Bard using httpx library for Async API calls.
1. __init__() -> __init__() is where the httpx AsyncClient is instantiated as part of the object and SESSION_HEADERS is being used here.
2. get_answer() -> get_answer() is async now, so will have to await the response from it.
3. _get_snim0e() -> _get_snim0e is also async because it is using httpx client to get data from Bard server, and will have to await the response from it.
4. _extract_links() -> No changes in this method, would not make any difference if it is not synchronous. 

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context
1. Giving a user more options by including both Async and Sync implementations.
2. Chatbots created using bardapi library can be much more efficient. 
<!--- If it fixes an open issue, please link to the issue here. -->

## How Has This Been Tested?
I tested it by manually running it on my local machine and checked the implementation of the functions.
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

dsdanielpark commented:
Dear, [nishantchauhan949](https://github.com/nishantchauhan949)

Thank you for your valuable contributions and excellent code! 

 I would like to provide a brief overview of the use cases and advantages of this code. Can you write simple example in readme.md? 

Thank you once again for your contribution. I would appreciate it if you could organize the code neatly in the script folder or any other appropriate location. 

If the update is not available after 24 hours, I will update it myself. 

Thank you in advance.

Regards,
Daniel.

nishantchauhan949 commented:
Dear, @dsdanielpark 

I have added a brief overview and a simple example in readme.md

Also, I have moved the core_async.py file to the translate_to directory. Hope that is fine.

PR: https://github.com/dsdanielpark/Bard-API/pull/48

Regards,
Nishant
